sonaranalyzer-dotnet/src/SonarAnalyzer.CSharp/Rules/ShouldImplementExportedInterfaces.cs

```diff
@@ -0,0 +1,142 @@
+                    {
+                        return;
+                    }
+
+                    var exportedType = GetExportedTypeSyntax(attributeSyntax, c.SemanticModel);
+                    var attributeTargetType = GetAttributeTargetSymbol(attributeSyntax, c.SemanticModel);
+
+                    if (exportedType != null &&
+                        attributeTargetType != null &&
+                        !attributeTargetType.DerivesOrImplements(exportedType))
+                    {
+                        c.ReportDiagnosticWhenActive(
+                            Diagnostic.Create(rule, attributeSyntax.GetLocation(), exportedType.Name, attributeTargetType.Name));
+                    }
+                },
+                SyntaxKind.Attribute);
+        }
+
+        private ITypeSymbol GetAttributeTargetSymbol(AttributeSyntax attributeSyntax, SemanticModel semanticModel)
+        {
+            // Parent is AttributeListSyntax, we handle only class attributes
+            var attributeTarget = attributeSyntax.Parent?.Parent as ClassDeclarationSyntax;
+            if (attributeTarget == null)
+            {
+                return null;
+            }
+            return semanticModel.GetDeclaredSymbol(attributeTarget);
+        }
+
+        private ITypeSymbol GetExportedTypeSyntax(AttributeSyntax attribute, SemanticModel semanticModel)
+        {
+            if (attribute.ArgumentList == null)
+            {
+                return null;
+            }
+
+            var arguments = attribute.ArgumentList.Arguments;
+            if (arguments.Count == 0 ||
+                arguments.Count > 2)
+            {
+                return null;
+            }
+
+            var typeOfExpression = GetTypeFromNamedArgument(arguments) ??
+                GetTypeFromSingleArgumentAttribute(arguments) ??
+                GetTypeFromDoubleArgumentAttribute(arguments, semanticModel);
+
+            var exportedTypeSyntax = (typeOfExpression as TypeOfExpressionSyntax)?.Type;
+            if (exportedTypeSyntax == null)
+            {
+                return null;
+            }
+
+            return semanticModel.GetSymbolInfo(exportedTypeSyntax).Symbol as ITypeSymbol;
+        }
+
+        private static ExpressionSyntax GetTypeFromNamedArgument(IEnumerable<AttributeArgumentSyntax> arguments) =>
+            arguments.FirstOrDefault(IsContractTypeNamedArgument)?.Expression;
+
+        private static ExpressionSyntax GetTypeFromDoubleArgumentAttribute(IEnumerable<AttributeArgumentSyntax> arguments,
+            SemanticModel semanticModel)
+        {
+            var firstArgument = arguments.ElementAtOrDefault(0)?.Expression;
+            if (firstArgument != null &&
+                semanticModel.GetConstantValue(firstArgument).Value is string)
+            {
+                // Two arguments, second should be typeof expression
+                return arguments.ElementAtOrDefault(1)?.Expression;

```

Comment:
What if it is not because the user as written invalid code?

Reply:
IF it is not, it will be null and the check will fail and nothing should be reported